docs/developer-guide/working-with-rules.md

```diff
@@ -311,6 +312,117 @@ Keep in mind that comments are technically not a part of the AST and are only at
 
 **Note:** One of the libraries adds AST node properties for comments - do not use these properties. Always use `sourceCode.getComments()` as this is the only guaranteed API for accessing comments (we will likely change how comments are handled later).
 
+### Accessing Code Paths
+
+ESLint is analyzing code paths while traversing AST. You can access that code path objects with `context.getCodePath()`.
+`context.getCodePath()` returns a `CodePath` object of the current location.
+
+> **Note:** That code path has not completed. If you want to get the completed code path, you can use events which will be described later.
+
+```ts
+/**
+ * CodePath objects exist for each function and the global.
+ * This has references of both the initial segment and the final segments.
+ * This is a directed graph.
+ */
+interface CodePath {
+    id: string;
+    upper: CodePath | null;
+    childCodePaths: CodePath[];
+    initialSeg: CodePathSegment;     // The initial segment.
+    finalSegs: CodePathSegment[];    // The final segments which includes both returned and thrown.
+    returnedSegs: CodePathSegment[]; // The final segments which includes only returned.
+    thrownSegs: CodePathSegment[];   // The final segments which includes only thrown.
+    currentSegs: CodePathSegment[];  // Segments of the current position.
+}
+```
+
+```ts
+/**
+ * This is a segment of a code path.
+ * This becomes unreachable preceded by one of `return`, `throw`, `break`, and `continue`.
+ */
+interface CodePathSegment {
+    id: string;
+    nextSegs: CodePathSegment[];
+    prevSegs: CodePathSegment[];
+    reachable: boolean;
+}
+```
+
```

Comment:
These are similar to `LinkedList<T>` and `LinkedListNode<T>`.
`CodePath` is references to the initial and the final of a code path.
`CodePathSegment` is parts of a code path.